src/bind.py

```diff
@@ -266,24 +266,44 @@ def _install_snap_package(
             logger.exception(error_msg)
             raise InstallError(error_msg) from e
 
-    def _zones_to_files_content(self, zones: list[models.Zone]) -> dict[str, str]:
+    def _zones_to_files_content(
+        self, zones: list[models.Zone], topology: models.Topology
+    ) -> dict[str, str]:
         """Return zone files and their content.
 
         Args:
             zones: list of the zones to transform to text
+            topology: Topology of the current deployment
 
         Returns:
             A dict whose keys are the domain of each zone
             and the values the content of the zone file
         """
         zone_files: dict[str, str] = {}
         for zone in zones:
-            content = constants.ZONE_HEADER_TEMPLATE.format(
+            content = constants.ZONE_APEX_TEMPLATE.format(
                 zone=zone.domain,
                 # The serial is the timestamp divided by 60.
                 # We only need precision to the minute and want to avoid overflows
                 serial=int(time.time() / 60),
             )
+
+            # By default, we hide the active unit.
+            # So only the standbies are used to respond to queries and receive NOTIFY events
+            ip_list: list[pydantic.IPvAnyAddress] = topology.standby_units_ip
+            # If we have no standby unit (a single unit deployment)
+            # then use all the units IPs instead
+            if not ip_list:
+                ip_list = topology.units_ip
```

Comment:
Minor/nitpick: I think using the current_unit_ip would make it more clear that is a single unit deployment

Reply:
I found it a bit more defensive to write it my way but I agree that it shouldn't make any difference in practice. So I changed it like you suggested.